dictionary.go

```diff
@@ -28,11 +28,11 @@ type Dictionary interface {
 	Len() int
 
 	// Returns the dictionary value at the given index.
-	Index(int) Value
+	Index(index int32) Value
 
-	// Inserts a value to the dictionary, returning the index at which it was
-	// recorded.
-	Insert(Value) int
+	// Inserts a value to the dictionary and writes the indexes at which values
+	// were inserted.
```

Comment:
I think the doc could be clearer... do you mean to say values has the same length as indexes? How big of an array does the caller need to pass in, in other words

Reply:
Oh... I'd sort of reverse the order of these I guess, so the input is first and the  value that is being returned is second.

Reply:
I think Go tends to encourage having the output parameter first, considering the signatures of `copy`, `io.Copy`, or sending values to channels with `ch <- v`?